### PR TITLE
[WIP] Rebased git rework

### PR DESCRIPTION
**Pull request by verythorough**
I was having trouble parsing the changes in #558, so I cherry-picked the Git workshop commits onto the current master.

### CODE REVIEW

**Comment by verythorough** [comment]
Deploy preview ready!

Built with commit 5f781606c69f94ef7f3b374ceb74217fea8a53b0

https://deploy-preview-560--teaching-materials.netlify.com

**Comment by verythorough** [comment]
I won't have time to update the marketing materials in this PR, so this is a reminder to file an issue for that.

File: git/exercise-sample-code/gallery.css [hunk 1]
```diff
@@ -0,0 +1 @@
+gallery.css
```
Comment (FrenchBen) [inline git/exercise-sample-code/gallery.css]: What's the point of this file?

File: git/exercise-sample-code/gallery.js [hunk 2]
```diff
@@ -0,0 +1 @@
+gallery.js
```
Comment (FrenchBen) [inline git/exercise-sample-code/gallery.js]: What's the point of this file?

File: git/exercise-sample-code/gallery.css (hunk 1 above)
Comment (hyendler) [reply git/exercise-sample-code/gallery.css]: was going to be used (so it was a place holder).  Removed.

File: git/exercise-sample-code/gallery.js (hunk 2 above)
Comment (hyendler) [reply git/exercise-sample-code/gallery.js]: was going to be used (so it was a place holder).  Removed.

**Pull request merged**